File: Algorithms/InsertionSort/InsertionSort.py
def insertion_sort(arr):
    for i in range(len(arr) - 1):
        for j in range(i + 1, 0, -1):
            if arr[j] < arr[j - 1]:
                arr[j], arr[j - 1] = arr[j - 1], arr[j]
            else:
                break

    return arr


if __name__ == "__main__":
    arr = [3, 4, 6, 2, 1]
    print(insertion_sort(arr))

File: Algorithms/InsertionSort/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    static int[] insertionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i + 1; j > 0; j--) {
                if (arr[j] < arr[j - 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j - 1];
                    arr[j - 1] = temp;
                } else {
                    break;
                }
            }
        }

        return arr;
    }

    public static void main(String[] args) {
        int[] arr = new int[] { 5, 4, 2, 3, 1 };
        int[] sortedArr = insertionSort(arr);
        System.out.println(Arrays.toString(sortedArr));
    }

}
